kv/txn_coord_sender.go

```diff
@@ -220,9 +244,62 @@ func NewTxnCoordSender(wrapped client.Sender, clock *hlc.Clock, linearizable boo
 		linearizable:      linearizable,
 		stopper:           stopper,
 	}
+
+	tc.stopper.RunWorker(tc.startStats)
 	return tc
 }
 
+// startStats blocks and periodically logs transaction statistics (throughput,
+// success rates, durations, ...).
+// TODO(mrtracy): Add this to TimeSeries.
+func (tc *TxnCoordSender) startStats() {
+	res := time.Millisecond // for duration logging resolution
+	lastNow := tc.clock.PhysicalNow()
+	for {
+		select {
+		case <-time.After(statusLogInterval):
+			if !log.V(1) {
+				continue
+			}
+
+			tc.Lock()
+			curStats := tc.txnStats
+			tc.txnStats = txnCoordStats{}
+			tc.Unlock()
+
+			now := tc.clock.PhysicalNow()
+
+			// Tests have weird clocks.
+			if now-lastNow <= 0 {
+				continue
+			}
+
+			num := len(curStats.durations)
+			dMax := time.Duration(stats.Max(curStats.durations))
+			dMean := time.Duration(stats.Mean(curStats.durations))
+			dDev := time.Duration(stats.StdDevP(curStats.durations))
+			rMax := stats.Max(curStats.restarts)
+			rMean := stats.Mean(curStats.restarts)
+			rDev := stats.StdDevP(curStats.restarts)
```

Comment:
Yeah, I agree that there is immediate value to this stuff. How about adding a TODO indicating that this needs to be generalized?